PulseEcoSwiftUI/PulseEcoSwiftUI/CityMap/SensorDetails/SenDetView.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  SenDetView.swift
+//  PulseEcoSwiftUI
+//
+//  Created by Maja Mitreska on 1/28/21.
+//  Copyright © 2021 Monika Dimitrova. All rights reserved.
+//
+
+import SwiftUI
+
+struct SenDetView: View {
+    @State var isExpanded: Bool = false
+    @EnvironmentObject var appVM: AppVM
+    @State var collapsedViewSize: CGSize = .zero
+    @State var expandedViewSize: CGSize = .zero
+    @EnvironmentObject var dataSource: DataSource
+    @EnvironmentObject var partialSheet : PartialSheetManager
+    @State var isSheetShown: PartialSheetState = .hidden
+    
+    var body: some View {
+        VStack {
+            RoundedRectangle(cornerRadius: CGFloat(5.0) / 2.0)
+                .frame(width: 40, height: 3.0)
+                .foregroundColor(Color(UIColor.systemGray2)).padding(.top, 10)
+            
+            
+            ChildSizeReader(size: self.$collapsedViewSize) {
+                
+                CollapsedView(viewModel: SensorDetailsVM(sensor: self.appVM.selectedSensor ?? SensorVM(),
+                                                         sensorsData: self.dataSource.sensorsData24h,
+                                                         selectedMeasure: self.dataSource.getCurrentMeasure(selectedMeasure: self.appVM.selectedMeasure)))
+                    .padding(.top, 5)
+                    .padding(.bottom, 40)
+                    .partialSheet(isPresented: self.$isSheetShown) {
+                        SensorDView(viewModel: ExpandedVM(sensorData24h: self.dataSource.sensorsData24h,
```

Comment:
Same here. Also you can extract this closure in a separate variable or method in this file and just reference it.